### PR TITLE
Update googleapis_auth README

### PR DESCRIPTION
**Pull request by nickmeinhold**
Adding a brief summary of the helpful discussion in #292 to googleapis_auth README

### CODE REVIEW

**Comment by nickmeinhold** [comment]
@MelbourneDeveloper if you have a second to check this that'd be great, thanks again the helpful write up

**Comment by kevmoo** [comment]
Sure, although this WHOLE readme needs to be fixed up. To use `async`/`await` etc – it's pretty bad...

**Comment by nickmeinhold** [comment]
> Sure, although this WHOLE readme needs to be fixed up. To use `async`/`await` etc – it's pretty bad...

Would you like me to have a go at that in this PR?  If so, are there any other specific improvements you can think of that should be made?

**Comment by kevmoo** [comment]
@nickmeinhold – you'd make my day!

Hrm. Just highlight the common use cases. Use `async`/`await`

Make sure the samples are actually valid code!

**Comment by MelbourneDeveloper** [comment]
@nickmeinhold I think that is helpful, but I'm not sure if it makes it clear to someone coming in fresh what is happening. The thing that people need to understand is that under normal circumstances, you would want to keep the client_secret secret but it's OK to leave it in the app (which holds it as text) for people to see as long as it doesn't need to be secret for other purposes. That's why you shouldn't reuse this client id for multiple purposes.

**Comment by nickmeinhold** [comment]
@MelbourneDeveloper thanks for that!  I guess my thinking was to let the link do the talking if people wanted further info - I had a go at updating the statement to get the important points you mentioned across.  Please feel free to suggest edits if that's easier than discussing, I'm very happy to just add whatever changes you suggest.

File: googleapis_auth/README.md [hunk 1]
```diff
@@ -142,7 +142,7 @@ void prompt(String url) {
 }
 ```
 
-The Client Id requires a `client_secret` here, but the OAuth2 flow used in this case [assumes that the app cannot keep secrets](https://developers.google.com/identity/protocols/oauth2/native-app). You should make sure not to re-use the same `client_secret` anywhere secrecy is required.
+The Client Id must be created with a `client_secret` here, however there is no way to properly secure a `client_secret` for installed/console applications. Fortunately the OAuth2 flow used in this case [assumes that the app cannot keep secrets](https://developers.google.com/identity/protocols/oauth2/native-app) so this particular `client_secret` does not need to be kept secret. You should however make sure not to re-use the same `client_secret` anywhere secrecy is required.
```
Comment (kevmoo) [inline googleapis_auth/README.md:145]:
```suggestion
The Client ID must be created with a `client_secret` here, however there is no way to properly secure a `client_secret` for installed/console applications. Fortunately the OAuth2 flow used in this case
[assumes that the app cannot keep secrets](https://developers.google.com/identity/protocols/oauth2/native-app) so this particular `client_secret` does not need to be kept secret. You should however make sure not to re-use the same `client_secret` anywhere secrecy is required.
```

**Review by kevmoo — APPROVED** [review]

**Comment by nickmeinhold** [comment]
@kevmoo (just FYI) I had marked the PR as a draft while I worked on the general improvements to the README you had suggested - I should have that done soon

**Comment by nickmeinhold** [comment]
- [x] Use `async`/`await`
- [x] Make sure the samples are actually valid code

> highlight the common use cases
  - I'm not sure what needs to be done here, could you say a bit more?

@kevmoo

**Comment by kevmoo** [comment]
@nickmeinhold – I did some follow-up cleanup https://github.com/google/googleapis.dart/pull/358/commits/3768218f54bc42514f109aa8f5ebed9ca5dc1926 FYI

**Comment by nickmeinhold** [comment]
@kevmoo looks much tidier 👍  and thanks for the FYI 🙂

**Pull request merged**